logger.go

```diff
@@ -85,6 +88,34 @@ func (logger Logger) ChildWithTags(name string, tags ...string) Logger {
 	return module.context.GetLogger(path, tags...)
 }
 
+// ChildWithLabels returns the Logger whose module name is the composed of this
+// Logger's name and the specified name with the correct associated labels.
+// Adding labels to the child logger will cause all child loggers to also
+// inherit the labels of the parent(s) loggers.
+// For targeting a singular logger with labels, use WithLabels which are not
+// inherited by child loggers.
```

Comment:
```suggestion
// ChildWithLabels returns the Logger whose module name is composed of this
// Logger's name along with specified name, copying this loggers labels.
// Adding labels to the child logger will cause all child loggers to also
// inherit the labels of the parent(s) loggers. Supplying a label that is already on this logger will
// overwrite the value with the supplied value.
// For targeting a singular logger with labels, use WithLabels which are not
// inherited by child loggers.
```

Reply:
Or maybe I'm misunderstanding, as I see there is 'labels' and 'impl.labels'.
Is it worth having that mix? It seems like it would make things more confusing to have 2 sets of labels, one which is inherited, and one which isn't.